src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt

```diff
@@ -60,8 +61,25 @@ class NotificationService(
 
         allPrisons.forEach { prison ->
             saveNotifications(
-                    csrClient.getShiftTaskNotifications(prison.csrPlanUnit, prison.region)
-                            .distinct())
+                   csrClient.getShiftTaskNotifications(prison.csrPlanUnit, prison.region)
+                            .map {
+                                if(ShiftNotificationType.from(it.shiftType) == ShiftNotificationType.SHIFT || it.shiftType.equals("task",true)) {
+                                    it.shiftType = ShiftNotificationType.SHIFT_TASK.value
+                                } else {
+                                    it.shiftType = ShiftNotificationType.OVERTIME_TASK.value
+                                }
+                                it
+                            }
+                            // There is a bug with the query that returns a Shift and an Overtime shift row for the same change.
+                            .groupBy { it.compoundKey() }
+                            .map { (_, value) ->
+                                if(value.any {ShiftNotificationType.from(it.shiftType) == ShiftNotificationType.OVERTIME_TASK }) {
```

Comment:
this would be a bit easier to follow if the assertion was declared as a variable. 
```

Reply:
you mean If(value.any { isOvertimeTask(it) } ?

Reply:
Yeah, either naming the lamda or naming the boolean eg

```
val hasOverTimeValue = value.any {ShiftNotificationType.from(it.shiftType) == ShiftNotificationType.OVERTIME_TASK }

if (hasOverTimeValue){
  ...
}

```

_DISCLAIMER: I may or may not be following what's happening here_